### PR TITLE
Remove macOS "borderless" recommendation

### PR DESCRIPTION
**Pull request by peppy**
As of SDL3, this is no longer a thing, and fullscreen should be the preferred execution mode.

Probably hold off merging this until we're sure that macOS isn't broken for others in this mode (I had issues locally, such as alt-tabbing being broken sooo...)

### CODE REVIEW

**Comment by koopafou** [comment]
I'm reading the SDL3 doc, you still have the choice to use exclusive mode (must be explicit) or borderless (default), it's not even marked as deprecated. For some reason the borderless option disappeared on Linux in the last update, and the remaining fullscreen option is using exclusive mode rather than being a secret borderless window to silently get rid of people's issues with alt-tabbing, you messed up something here.

**Comment by bdach** [comment]
- https://github.com/ppy/osu-framework/pull/6264
- https://github.com/ppy/osu-framework/pull/6426

probably gonna link these in here before the above point of discussion gets too heated (started off pretty heated already)

**Comment by koopafou** [comment]
Sorry I wasn't trying to get the discussion heated, I was pointing out that maybe the issue came from the selection of the fullscreen mode rather than trying to handle user-facing issues that aren't even an osu or SDL problem. Fixing exclusive mode alt-tab is a lost cause in modern days, it's not even a macOS thing, it also occurs on Windows and Linux in any video game because the hardware is designed to force it that way. So yeah while you could talk about the performance improvements and be technically true, in practice window managers should be fast enough that the performance hit of compositing is probably placebo. But even if you prove it's placebo, some players will still want legacy fullscreen mode, and you can't remove the borderless fullscreen mode that just works. So might as well keep both and tell people to go borderless if they have issues.

File: osu.Game/Overlays/Settings/Sections/Graphics/LayoutSettings.cs [hunk 1]
```diff
@@ -269,14 +269,7 @@ private void updateDisplaySettingsVisibility()
         private void updateScreenModeWarning()
         {
             if (RuntimeInfo.OS == RuntimeInfo.Platform.macOS)
-            {
-                if (windowModeDropdown.Current.Value == WindowMode.Fullscreen)
```
[the remark this answers is not in the document]
Comment (peppy) [reply osu.Game/Overlays/Settings/Sections/Graphics/LayoutSettings.cs:273]: Sure, applied.

File: osu.Game/Overlays/Settings/Sections/Graphics/LayoutSettings.cs [hunk 2]
```diff
@@ -268,7 +268,8 @@ private void updateDisplaySettingsVisibility()
 
         private void updateScreenModeWarning()
         {
-            if (RuntimeInfo.OS == RuntimeInfo.Platform.macOS)
+            // Can be removed once we stop supporting SDL2.
+            if (!FrameworkEnvironment.UseSDL3)
```
Comment (Joehuu) [inline osu.Game/Overlays/Settings/Sections/Graphics/LayoutSettings.cs:272]: This still needs the macOS condition.

**Review by frenzibyte — APPROVED** [review]

**Comment by frenzibyte** [comment]
I think this can still go in anyway since all of what the note highlights is not a thing in any of SDL3's fullscreen modes.

**Pull request merged**